cl/lib/pacer.py

```diff
@@ -530,3 +534,118 @@ def normalize_attorney_contact(c, fallback_name=""):
     address_info = normalize_address_info(dict(address_info))
     address_info["lookup_key"] = make_address_lookup_key(address_info)
     return address_info, atty_info
+
+
+class ConnectionType(TypedDict):
+    connection_ok: bool
+    status_code: int | None
+    date_time: datetime
+
+
+def check_pacer_court_connectivity(court_id: str) -> ConnectionType:
+    """Check if PACER is accessible for the given court.
+
+    :param court_id: The court ID to check.
+    :returns: A dict with the court connection status.
+    """
+
+    url = f"https://ecf.{court_id}.uscourts.gov/"
+
+    connection_ok = False
+    status_code = None
+    try:
+        r = requests.get(url, timeout=5)
+        if r.status_code == requests.codes.ok:
+            connection_ok = True
+        status_code = r.status_code
+    except requests.exceptions.RequestException as e:
+        connection_ok = False
+
+    blocked_dict: ConnectionType = {
+        "connection_ok": connection_ok,
+        "status_code": status_code,
+        "date_time": datetime.now(timezone.utc),
+    }
+    return blocked_dict
+
+
+def get_or_cache_court_status(court_id: str, server_ip: str) -> bool:
+    """Get the court status from Redis or cache it if it's not there.
+
+    :param court_id: The court ID to check.
+    :param server_ip: The server IP address.
+    :return: True if connection was successful, False otherwise.
+    """
+
+    court_status_key = "status:pacer:court.%s:ip.%s"
+    r = make_redis_interface("CACHE", decode_responses=False)
+    pickle_status = r.get(court_status_key % (court_id, server_ip))
+    if pickle_status:
+        court_status = pickle.loads(pickle_status)
+        return court_status
+
+    # Unable to find court_status in cache, getting it from request.
+    connection_info = check_pacer_court_connectivity(court_id)
+    current_status = connection_info["connection_ok"]
+
+    # Stores court connection status with court ID and server IP as key.
+    # 10 minutes expiration time.
+    status_expiration = 60 * 10
```

Comment:
I think I'd reduce this to one minute, so we don't stay blocked longer than necessary. One extra request against their server per minute ain't bad. Even reducing this to 30s seems OK, no?

Reply:
Sure, I've set it to 30s